src/main/java/org/opensearch/sdk/SDKHttpRequest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class helps to get instance of HttpRequest
+ */
+public class SDKHttpRequest implements HttpRequest {
+    private final RestRequest.Method method;
+    private final String uri;
+    private final BytesReference content;
+    private final Map<String, List<String>> headers;
+    private final HttpVersion httpVersion;
+
+    /**
+    * Instantiates this class with a copy of request
+    *
+    * @param request The request
+    */
+    public SDKHttpRequest(ExtensionRestRequest request) {
+        this.method = request.method();
+        this.uri = request.uri();
+        this.content = request.content();
+        this.headers = request.headers();
+        this.httpVersion = request.httpVersion();
+    }
+
+    /**
+    * This method returns request method
+    * @return A method of request
+    */
```

Comment:
You do not need to include any javadocs when there is an `@Override` annotation, as it inherits the documentation from the superclass.  

Delete this javadoc, and all the ones which are overrides that actually return data.  If you want to include a javadoc for the overrides that return null, you should indicate that instead of pretending it does something. 

Something like this, perhaps.
```java
/**
 * Not implemented.
 * @return null
 */
```
or for the `releaseAndCopy()` methods:
```java
/**
 * Not implemented. Does nothing.
 * @return null
 */
```